tdp/cli/commands/deploy.py

```diff
@@ -86,72 +38,58 @@ def deploy(
-        set_nodes.update(targets)
-    set_difference = set_nodes.difference(dag.operations)
-    if set_difference:
-        raise click.BadParameter(f"{set_difference} are not valid nodes")
-    run_directory = run_directory.absolute() if run_directory else None
+        raise click.BadParameter(f"{vars} does not exist.")
+
+    cluster_variables = ClusterVariables.get_cluster_variables(
+        collections, vars, validate=validate
+    )
+    check_services_cleanliness(cluster_variables)
 
-    ansible_executor = AnsibleExecutor(
-        run_directory=run_directory,
-        dry=dry or mock_deploy,
+    deployment_runner = DeploymentRunner(
+        collections,
+        AnsibleExecutor(
+            run_directory=run_directory.absolute() if run_directory else None,
+            dry=dry or mock_deploy,
+        ),
+        cluster_variables,
     )
+
     session_class = get_session_class(database_dsn)
     with session_class() as session:
-        cluster_variables = ClusterVariables.get_cluster_variables(
-            collections, vars, validate=validate
-        )
-        check_services_cleanliness(cluster_variables)
-
-        deployment_runner = DeploymentRunner(
-            collections, ansible_executor, cluster_variables
-        )
-        if sources:
-            click.echo(f"Deploying from {sources}")
-        elif targets:
-            click.echo(f"Deploying to {targets}")
-        else:
-            click.echo(f"Deploying TDP")
-        try:
-            deployment_plan = DeploymentPlan.from_dag(
-                dag,
-                sources=sources,
-                targets=targets,
-                filter_expression=filter,
-                filter_type=filter_type,
-                restart=restart,
+        previous_deployment_log = get_planned_deployment_log(session)
+        if previous_deployment_log is None:
+            raise click.ClickException(
+                "No previous deployment found, please run `tdp plan` first."
             )
-        except Exception as e:
-            raise click.ClickException(str(e)) from e
+        deployment_plan = DeploymentPlan.from_deployment_log(
+            collections, previous_deployment_log
+        )
         deployment_iterator = deployment_runner.run(deployment_plan)
         if dry:
             for _ in deployment_iterator:
                 pass
         else:
-            session.add(deployment_iterator.log)
-            # insert pending deployment log
+            # Update deployment log to RUNNING
+            session.merge(deployment_iterator.deployment_log)
             session.commit()
-            for operation_log, service_component_log in deployment_iterator:
+            for index, (operation_log, service_component_log) in enumerate(
+                deployment_iterator
+            ):
                 if operation_log is not None:
-                    session.add(operation_log)
+                    operation_log.order = index
```

Comment:
Why the order is set here ? The `OperationLog` returned by the iterator should be ready to put in the database without modification.

The order is already defined when the operation is store in the database by the `plan`.